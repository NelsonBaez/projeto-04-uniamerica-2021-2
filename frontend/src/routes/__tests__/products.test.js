import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import Products from '../products';

const server = setupServer(
  rest.get('http://localhost:8080/products', (req, res, ctx) => {
    return res(ctx.json([
      {
        name : "caneta",
        defaultPrice:"10",
        productType: "1",
      }
    ]))
  }),
)

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

test("renders products", async () => {
  
  render(
    <MemoryRouter><Products /></MemoryRouter>
  )
  const element = await screen.findByText('caneta');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent('caneta');
})

test('matches snapshot', () => {
  const tree = renderer.create(
    
    <MemoryRouter><Products /></MemoryRouter>
  
  ).toJSON();
  expect(tree).toMatchSnapshot();
  
});