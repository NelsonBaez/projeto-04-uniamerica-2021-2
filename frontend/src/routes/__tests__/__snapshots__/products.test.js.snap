// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main>
  <h2
    className="font-bold my-5 text-3xl"
  >
    Produtos
  </h2>
  <div
    className="flex flex-row flex-wrap container text-center mx-auto shadow-md"
  >
    <div
      className="p-5 border-r-2 text-xl bg-green-200"
    >
      <div>
        <a
          className="text-blue-400"
          data-testid="link-0"
          href="/products/new"
          onClick={[Function]}
        >
          Novo Produto
        </a>
      </div>
    </div>
    <div
      className="flex-1 p-5 bg-gray-100"
    />
  </div>
</main>
`;
